cmd/pilot-agent/main.go

```diff
@@ -47,60 +56,87 @@ var (
 		Short: "Envoy proxy agent",
 		RunE: func(c *cobra.Command, args []string) error {
 			// set values from environment variables
-			if sidecar.IPAddress == "" {
-				sidecar.IPAddress = os.Getenv("INSTANCE_IP")
-			}
-			if sidecar.ID == "" {
-				sidecar.ID = os.Getenv("POD_NAME") + "." + os.Getenv("POD_NAMESPACE")
-			}
-			// receive mesh configuration
-			mesh, err := cmd.ReadMeshConfig(meshconfig)
-			if err != nil {
-				return multierror.Prefix(err, "failed to read mesh configuration.")
-			}
+			if serviceregistry == proxy.KubernetesRegistry {
```

Comment:
Why ? The env variables should work without any k8s registry - the name 'pod' and instance can be used without k8s :-)